{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns \n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filenames\n",
    "\n",
    "#MAF protein coding files\n",
    "MAF_VEP_protein_tumour_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_protein_coding_tumour.csv'\n",
    "MAF_VEP_protein_normal_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_protein_coding_normal.csv'\n",
    "\n",
    "#Gene expression FPKM files\n",
    "FPKM_tumour_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_FPKM_tumour.csv'\n",
    "FPKM_normal_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_FPKM_normal.csv'\n",
    "\n",
    "#eQTL files\n",
    "cis_eQTL_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/eQTL/COAD_tumor.cis_eQTL.txt'\n",
    "trans_eQTL_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/eQTL/COAD_tumor.trans_eQTL.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load cis_eQTL and trans_eQTL files\n",
    "\n",
    "cis_eQTL = pd.read_csv(cis_eQTL_file,sep='\\t')\n",
    "trans_eQTL = pd.read_csv(trans_eQTL_file,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>gene</th>\n",
       "      <th>beta</th>\n",
       "      <th>t-stat</th>\n",
       "      <th>p-value</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs6531717:39454321:G:A</td>\n",
       "      <td>RPL9|6133</td>\n",
       "      <td>-1.138911</td>\n",
       "      <td>-33.024236</td>\n",
       "      <td>3.429985e-95</td>\n",
       "      <td>1.639675e-87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2687969:39452499:T:C</td>\n",
       "      <td>RPL9|6133</td>\n",
       "      <td>-1.135399</td>\n",
       "      <td>-32.763508</td>\n",
       "      <td>1.818892e-94</td>\n",
       "      <td>4.347531e-87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2249563:39455378:A:G</td>\n",
       "      <td>RPL9|6133</td>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2687955:39457153:C:T</td>\n",
       "      <td>RPL9|6133</td>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2687956:39457430:T:C</td>\n",
       "      <td>RPL9|6133</td>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      SNP       gene      beta     t-stat       p-value  \\\n",
       "0  rs6531717:39454321:G:A  RPL9|6133 -1.138911 -33.024236  3.429985e-95   \n",
       "1  rs2687969:39452499:T:C  RPL9|6133 -1.135399 -32.763508  1.818892e-94   \n",
       "2  rs2249563:39455378:A:G  RPL9|6133 -1.134352 -31.943949  3.633020e-92   \n",
       "3  rs2687955:39457153:C:T  RPL9|6133 -1.134352 -31.943949  3.633020e-92   \n",
       "4  rs2687956:39457430:T:C  RPL9|6133 -1.134352 -31.943949  3.633020e-92   \n",
       "\n",
       "            FDR  \n",
       "0  1.639675e-87  \n",
       "1  4.347531e-87  \n",
       "2  2.481049e-85  \n",
       "3  2.481049e-85  \n",
       "4  2.481049e-85  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "cis_eQTL.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new SNP columsn by splitting the SNP column and also remove numbers from gene names\n",
    "\n",
    "cis_eQTL['Gene_symbol'] = cis_eQTL['gene'].apply(lambda gene: gene.split('|')[0])\n",
    "\n",
    "cis_eQTL['cis_SNP_ID'] = cis_eQTL['SNP'].apply(lambda x: x.split(':')[0])\n",
    "cis_eQTL['SNP_Position_Chr (hg19)'] = cis_eQTL['SNP'].apply(lambda x: x.split(':')[1])\n",
    "cis_eQTL['SNP_Allele'] = cis_eQTL['SNP'].apply(lambda x: x.split(':')[2] + '/' + x.split(':')[3])\n",
    "\n",
    "#Drop old SNP and gene col\n",
    "cis_eQTL.drop(['SNP','gene'],axis =1 ,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>beta</th>\n",
       "      <th>t-stat</th>\n",
       "      <th>p-value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>cis_SNP_ID</th>\n",
       "      <th>SNP_Position_Chr (hg19)</th>\n",
       "      <th>SNP_Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>-1.138911</td>\n",
       "      <td>-33.024236</td>\n",
       "      <td>3.429985e-95</td>\n",
       "      <td>1.639675e-87</td>\n",
       "      <td>RPL9</td>\n",
       "      <td>rs6531717</td>\n",
       "      <td>39454321</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-1.135399</td>\n",
       "      <td>-32.763508</td>\n",
       "      <td>1.818892e-94</td>\n",
       "      <td>4.347531e-87</td>\n",
       "      <td>RPL9</td>\n",
       "      <td>rs2687969</td>\n",
       "      <td>39452499</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "      <td>RPL9</td>\n",
       "      <td>rs2249563</td>\n",
       "      <td>39455378</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "      <td>RPL9</td>\n",
       "      <td>rs2687955</td>\n",
       "      <td>39457153</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-1.134352</td>\n",
       "      <td>-31.943949</td>\n",
       "      <td>3.633020e-92</td>\n",
       "      <td>2.481049e-85</td>\n",
       "      <td>RPL9</td>\n",
       "      <td>rs2687956</td>\n",
       "      <td>39457430</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       beta     t-stat       p-value           FDR Gene_symbol cis_SNP_ID  \\\n",
       "0 -1.138911 -33.024236  3.429985e-95  1.639675e-87        RPL9  rs6531717   \n",
       "1 -1.135399 -32.763508  1.818892e-94  4.347531e-87        RPL9  rs2687969   \n",
       "2 -1.134352 -31.943949  3.633020e-92  2.481049e-85        RPL9  rs2249563   \n",
       "3 -1.134352 -31.943949  3.633020e-92  2.481049e-85        RPL9  rs2687955   \n",
       "4 -1.134352 -31.943949  3.633020e-92  2.481049e-85        RPL9  rs2687956   \n",
       "\n",
       "  SNP_Position_Chr (hg19) SNP_Allele  \n",
       "0                39454321        G/A  \n",
       "1                39452499        T/C  \n",
       "2                39455378        A/G  \n",
       "3                39457153        C/T  \n",
       "4                39457430        T/C  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cis_eQTL.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Now repeat process for the trans_eQTLS\n",
    "\n",
    "trans_eQTL['Gene_symbol'] = trans_eQTL['gene'].apply(lambda gene: gene.split('|')[0])\n",
    "\n",
    "trans_eQTL['trans_SNP_ID'] = trans_eQTL['SNP'].apply(lambda x: x.split(':')[0])\n",
    "trans_eQTL['SNP_Position_Chr (hg19)'] = trans_eQTL['SNP'].apply(lambda x: x.split(':')[1])\n",
    "trans_eQTL['SNP_Allele'] = trans_eQTL['SNP'].apply(lambda x: x.split(':')[2] + '/' + x.split(':')[3])\n",
    "\n",
    "#Drop old SNP and gene col\n",
    "trans_eQTL.drop(['SNP','gene'],axis =1 ,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>beta</th>\n",
       "      <th>t-stat</th>\n",
       "      <th>p-value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>trans_SNP_ID</th>\n",
       "      <th>SNP_Position_Chr (hg19)</th>\n",
       "      <th>SNP_Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.919773</td>\n",
       "      <td>15.811843</td>\n",
       "      <td>5.858615e-40</td>\n",
       "      <td>4.655876e-29</td>\n",
       "      <td>HSP90AB2P</td>\n",
       "      <td>rs13296</td>\n",
       "      <td>44218120</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.994873</td>\n",
       "      <td>15.474549</td>\n",
       "      <td>9.019558e-39</td>\n",
       "      <td>3.583948e-28</td>\n",
       "      <td>LOC388955</td>\n",
       "      <td>rs28580074</td>\n",
       "      <td>176721198</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.984245</td>\n",
       "      <td>15.239219</td>\n",
       "      <td>6.069065e-38</td>\n",
       "      <td>1.607707e-27</td>\n",
       "      <td>LOC388955</td>\n",
       "      <td>rs28684504</td>\n",
       "      <td>176727438</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.865920</td>\n",
       "      <td>15.096688</td>\n",
       "      <td>1.924334e-37</td>\n",
       "      <td>3.659264e-27</td>\n",
       "      <td>POM121L10P</td>\n",
       "      <td>rs140507</td>\n",
       "      <td>23646512</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.865200</td>\n",
       "      <td>15.074532</td>\n",
       "      <td>2.302276e-37</td>\n",
       "      <td>3.659264e-27</td>\n",
       "      <td>POM121L10P</td>\n",
       "      <td>rs131684</td>\n",
       "      <td>23646257</td>\n",
       "      <td>A/T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       beta     t-stat       p-value           FDR Gene_symbol trans_SNP_ID  \\\n",
       "0  0.919773  15.811843  5.858615e-40  4.655876e-29   HSP90AB2P      rs13296   \n",
       "1  0.994873  15.474549  9.019558e-39  3.583948e-28   LOC388955   rs28580074   \n",
       "2  0.984245  15.239219  6.069065e-38  1.607707e-27   LOC388955   rs28684504   \n",
       "3  0.865920  15.096688  1.924334e-37  3.659264e-27  POM121L10P     rs140507   \n",
       "4  0.865200  15.074532  2.302276e-37  3.659264e-27  POM121L10P     rs131684   \n",
       "\n",
       "  SNP_Position_Chr (hg19) SNP_Allele  \n",
       "0                44218120        A/G  \n",
       "1               176721198        T/C  \n",
       "2               176727438        T/C  \n",
       "3                23646512        C/T  \n",
       "4                23646257        A/T  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_eQTL.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CCDC74B      384\n",
       "LOC285359    287\n",
       "ZNF322B      273\n",
       "INTS4L1      147\n",
       "LOC647288    116\n",
       "            ... \n",
       "CRNKL1         1\n",
       "PTPN9          1\n",
       "EIF3J          1\n",
       "DLGAP2         1\n",
       "WASH5P         1\n",
       "Name: Gene_symbol, Length: 373, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_eQTL['Gene_symbol'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "LOC399744    1638\n",
       "FABP2        1076\n",
       "SNHG5        1039\n",
       "ZNF117        998\n",
       "XKR9          980\n",
       "             ... \n",
       "DDX39           1\n",
       "NELL2           1\n",
       "CDC42EP3        1\n",
       "BCHE            1\n",
       "ADH5            1\n",
       "Name: Gene_symbol, Length: 5048, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cis_eQTL['Gene_symbol'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
