{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns \n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Filenames\n",
    "\n",
    "#MAF protein coding files\n",
    "MAF_VEP_protein_tumour_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_protein_coding_tumour.csv'\n",
    "MAF_VEP_protein_normal_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_protein_coding_normal.csv'\n",
    "\n",
    "#Gene expression FPKM files\n",
    "FPKM_tumour_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_FPKM_tumour.csv'\n",
    "FPKM_normal_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/FPKM_MAF_notebook/COAD_FPKM_normal.csv'\n",
    "\n",
    "#Isoform switch file\n",
    "isoform_switches_file = '/Users/jake/OneDrive - University of Glasgow/Project/COAD/isoform_switch/coad-isoform-switch-iktspmodel.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load isoform switches\n",
    "isoform_switches = pd.read_csv(isoform_switches_file,sep='\\t',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBLN2|2199,uc011avb.1</td>\n",
       "      <td>FBLN2|2199,uc011ava.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CD44|960,uc001mvw.2</td>\n",
       "      <td>CD44|960,uc001mvx.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC39A14|23516,uc003xbq.3</td>\n",
       "      <td>SLC39A14|23516,uc003xbp.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UGP2|7360,uc002scl.2</td>\n",
       "      <td>UGP2|7360,uc002scm.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IL1R2|7850,uc002tbn.2</td>\n",
       "      <td>IL1R2|7850,uc002tbm.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           0                          1\n",
       "0      FBLN2|2199,uc011avb.1      FBLN2|2199,uc011ava.1\n",
       "1        CD44|960,uc001mvw.2        CD44|960,uc001mvx.2\n",
       "2  SLC39A14|23516,uc003xbq.3  SLC39A14|23516,uc003xbp.3\n",
       "3       UGP2|7360,uc002scl.2       UGP2|7360,uc002scm.2\n",
       "4      IL1R2|7850,uc002tbn.2      IL1R2|7850,uc002tbm.2"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_switches.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new cols for isoform switches, with one displaying gene_symbol and the other two displaying the two transcript IDs\n",
    "\n",
    "isoform_switches['Gene_symbol'] = isoform_switches[0].apply(lambda x: x.split('|')[0])\n",
    "isoform_switches['Transcript_Id_1'] = isoform_switches[0].apply(lambda x: x.split(',')[1])\n",
    "isoform_switches['Transcript_Id_2'] = isoform_switches[1].apply(lambda x: x.split(',')[1])\n",
    "\n",
    "#Drop the initial columns\n",
    "isoform_switches.drop([0,1],axis=1,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>Transcript_Id_1</th>\n",
       "      <th>Transcript_Id_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FBLN2</td>\n",
       "      <td>uc011avb.1</td>\n",
       "      <td>uc011ava.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CD44</td>\n",
       "      <td>uc001mvw.2</td>\n",
       "      <td>uc001mvx.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SLC39A14</td>\n",
       "      <td>uc003xbq.3</td>\n",
       "      <td>uc003xbp.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UGP2</td>\n",
       "      <td>uc002scl.2</td>\n",
       "      <td>uc002scm.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IL1R2</td>\n",
       "      <td>uc002tbn.2</td>\n",
       "      <td>uc002tbm.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene_symbol Transcript_Id_1 Transcript_Id_2\n",
       "0       FBLN2      uc011avb.1      uc011ava.1\n",
       "1        CD44      uc001mvw.2      uc001mvx.2\n",
       "2    SLC39A14      uc003xbq.3      uc003xbp.3\n",
       "3        UGP2      uc002scl.2      uc002scm.2\n",
       "4       IL1R2      uc002tbn.2      uc002tbm.2"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_switches.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Number of genes associated with isoform switches\n",
    "\n",
    "isoform_switches['Gene_symbol'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Filter for these genes in the FPKM gene expression and MAF protein coding dataframes ####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Load files\n",
    "\n",
    "MAF_VEP_protein_tumour = pd.read_csv(MAF_VEP_protein_tumour_file,low_memory=False,index_col=0)\n",
    "MAF_VEP_protein_normal = pd.read_csv(MAF_VEP_protein_normal_file,low_memory=False,index_col=0)\n",
    "\n",
    "FPKM_tumour = pd.read_csv(FPKM_tumour_file,index_col=0)\n",
    "FPKM_normal = pd.read_csv(FPKM_normal_file,index_col=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#First filter the FPKM tumour and then normal dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check if any present in isoform_switches not found in FPKM\n",
    "#Could be due to differences in gene label => add manualy to FPKM_iso if so\n",
    "\n",
    "set(isoform_switches['Gene_symbol']) - set(FPKM_tumour['Gene_symbol']) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create list of iso_form genes to filter for, including any identified above with alternative names\n",
    "\n",
    "isoform_genes_FPKM = list(isoform_switches['Gene_symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FPKM tumour filter for isoform switch genes\n",
    "\n",
    "FPKM_tumour_iso = FPKM_tumour.loc[FPKM_tumour['Gene_symbol'].isin(isoform_genes_FPKM)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ensembl_id</th>\n",
       "      <th>Gene_symbol</th>\n",
       "      <th>TCGA-AA-3867</th>\n",
       "      <th>TCGA-CA-6719</th>\n",
       "      <th>TCGA-NH-A50V</th>\n",
       "      <th>TCGA-AA-A01C</th>\n",
       "      <th>TCGA-AA-A02F</th>\n",
       "      <th>TCGA-AA-A03F</th>\n",
       "      <th>TCGA-AZ-4615</th>\n",
       "      <th>TCGA-AA-3854</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-A6-6138</th>\n",
       "      <th>TCGA-A6-2685</th>\n",
       "      <th>TCGA-QG-A5YX</th>\n",
       "      <th>TCGA-AY-A54L</th>\n",
       "      <th>TCGA-A6-2671</th>\n",
       "      <th>TCGA-NH-A8F8</th>\n",
       "      <th>TCGA-CM-5863</th>\n",
       "      <th>TCGA-A6-3809.2</th>\n",
       "      <th>TCGA-D5-6932</th>\n",
       "      <th>TCGA-AA-3675</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>ENSG00000026508</td>\n",
       "      <td>CD44</td>\n",
       "      <td>49.437336</td>\n",
       "      <td>26.181544</td>\n",
       "      <td>43.585652</td>\n",
       "      <td>25.303669</td>\n",
       "      <td>57.488870</td>\n",
       "      <td>22.468272</td>\n",
       "      <td>46.933128</td>\n",
       "      <td>56.347290</td>\n",
       "      <td>...</td>\n",
       "      <td>34.361535</td>\n",
       "      <td>32.096380</td>\n",
       "      <td>29.925135</td>\n",
       "      <td>32.409770</td>\n",
       "      <td>31.520747</td>\n",
       "      <td>50.971634</td>\n",
       "      <td>42.929245</td>\n",
       "      <td>59.984435</td>\n",
       "      <td>50.069357</td>\n",
       "      <td>59.563262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1287</th>\n",
       "      <td>ENSG00000075142</td>\n",
       "      <td>SRI</td>\n",
       "      <td>44.841315</td>\n",
       "      <td>38.171138</td>\n",
       "      <td>24.658549</td>\n",
       "      <td>43.792638</td>\n",
       "      <td>27.200675</td>\n",
       "      <td>15.084670</td>\n",
       "      <td>17.863711</td>\n",
       "      <td>114.546805</td>\n",
       "      <td>...</td>\n",
       "      <td>37.183063</td>\n",
       "      <td>47.035315</td>\n",
       "      <td>48.634665</td>\n",
       "      <td>60.581531</td>\n",
       "      <td>28.211444</td>\n",
       "      <td>23.956176</td>\n",
       "      <td>56.181913</td>\n",
       "      <td>25.066261</td>\n",
       "      <td>22.739080</td>\n",
       "      <td>36.772185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1667</th>\n",
       "      <td>ENSG00000085733</td>\n",
       "      <td>CTTN</td>\n",
       "      <td>43.419941</td>\n",
       "      <td>31.008285</td>\n",
       "      <td>49.971597</td>\n",
       "      <td>43.526294</td>\n",
       "      <td>23.395513</td>\n",
       "      <td>70.448546</td>\n",
       "      <td>30.153891</td>\n",
       "      <td>44.858901</td>\n",
       "      <td>...</td>\n",
       "      <td>29.078803</td>\n",
       "      <td>38.652049</td>\n",
       "      <td>43.775762</td>\n",
       "      <td>32.808479</td>\n",
       "      <td>32.837844</td>\n",
       "      <td>42.640666</td>\n",
       "      <td>32.527568</td>\n",
       "      <td>39.002346</td>\n",
       "      <td>47.686444</td>\n",
       "      <td>21.455800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2940</th>\n",
       "      <td>ENSG00000104635</td>\n",
       "      <td>SLC39A14</td>\n",
       "      <td>22.918339</td>\n",
       "      <td>16.831717</td>\n",
       "      <td>12.825033</td>\n",
       "      <td>20.000358</td>\n",
       "      <td>20.654395</td>\n",
       "      <td>12.901203</td>\n",
       "      <td>59.061575</td>\n",
       "      <td>23.314894</td>\n",
       "      <td>...</td>\n",
       "      <td>48.667234</td>\n",
       "      <td>22.855138</td>\n",
       "      <td>34.125630</td>\n",
       "      <td>22.872968</td>\n",
       "      <td>7.496100</td>\n",
       "      <td>21.732310</td>\n",
       "      <td>32.023106</td>\n",
       "      <td>25.836310</td>\n",
       "      <td>18.235208</td>\n",
       "      <td>27.093669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4452</th>\n",
       "      <td>ENSG00000115590</td>\n",
       "      <td>IL1R2</td>\n",
       "      <td>1.729770</td>\n",
       "      <td>1.097672</td>\n",
       "      <td>0.570912</td>\n",
       "      <td>0.479973</td>\n",
       "      <td>0.822586</td>\n",
       "      <td>2.869665</td>\n",
       "      <td>4.744338</td>\n",
       "      <td>31.282369</td>\n",
       "      <td>...</td>\n",
       "      <td>2.068797</td>\n",
       "      <td>2.393464</td>\n",
       "      <td>14.959713</td>\n",
       "      <td>0.994161</td>\n",
       "      <td>1.415372</td>\n",
       "      <td>0.253236</td>\n",
       "      <td>8.802575</td>\n",
       "      <td>2.729988</td>\n",
       "      <td>1.729465</td>\n",
       "      <td>1.220621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6093</th>\n",
       "      <td>ENSG00000129009</td>\n",
       "      <td>ISLR</td>\n",
       "      <td>69.811483</td>\n",
       "      <td>42.516880</td>\n",
       "      <td>51.739143</td>\n",
       "      <td>58.313480</td>\n",
       "      <td>10.410286</td>\n",
       "      <td>8.693111</td>\n",
       "      <td>27.509687</td>\n",
       "      <td>5.299220</td>\n",
       "      <td>...</td>\n",
       "      <td>26.867733</td>\n",
       "      <td>71.743560</td>\n",
       "      <td>7.693835</td>\n",
       "      <td>1.601303</td>\n",
       "      <td>49.215278</td>\n",
       "      <td>37.554881</td>\n",
       "      <td>21.229201</td>\n",
       "      <td>2.308740</td>\n",
       "      <td>30.980162</td>\n",
       "      <td>16.907314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6396</th>\n",
       "      <td>ENSG00000131089</td>\n",
       "      <td>ARHGEF9</td>\n",
       "      <td>2.116260</td>\n",
       "      <td>2.724855</td>\n",
       "      <td>2.982264</td>\n",
       "      <td>2.773046</td>\n",
       "      <td>2.261290</td>\n",
       "      <td>2.707335</td>\n",
       "      <td>2.139916</td>\n",
       "      <td>2.228208</td>\n",
       "      <td>...</td>\n",
       "      <td>2.665840</td>\n",
       "      <td>3.260938</td>\n",
       "      <td>3.532286</td>\n",
       "      <td>3.843074</td>\n",
       "      <td>2.326598</td>\n",
       "      <td>2.873641</td>\n",
       "      <td>2.192778</td>\n",
       "      <td>1.684769</td>\n",
       "      <td>2.963232</td>\n",
       "      <td>2.286335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6750</th>\n",
       "      <td>ENSG00000133392</td>\n",
       "      <td>MYH11</td>\n",
       "      <td>24.526326</td>\n",
       "      <td>60.857761</td>\n",
       "      <td>5.664525</td>\n",
       "      <td>15.126343</td>\n",
       "      <td>6.824040</td>\n",
       "      <td>3.769247</td>\n",
       "      <td>2.303767</td>\n",
       "      <td>1.568848</td>\n",
       "      <td>...</td>\n",
       "      <td>12.369897</td>\n",
       "      <td>138.054493</td>\n",
       "      <td>1.350113</td>\n",
       "      <td>0.849897</td>\n",
       "      <td>25.909564</td>\n",
       "      <td>52.773071</td>\n",
       "      <td>19.038967</td>\n",
       "      <td>0.888296</td>\n",
       "      <td>18.184051</td>\n",
       "      <td>1.436682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8483</th>\n",
       "      <td>ENSG00000143882</td>\n",
       "      <td>ATP6V1C2</td>\n",
       "      <td>2.374979</td>\n",
       "      <td>1.542474</td>\n",
       "      <td>1.674405</td>\n",
       "      <td>4.851438</td>\n",
       "      <td>2.605318</td>\n",
       "      <td>0.851872</td>\n",
       "      <td>0.524931</td>\n",
       "      <td>0.643164</td>\n",
       "      <td>...</td>\n",
       "      <td>1.414512</td>\n",
       "      <td>5.900615</td>\n",
       "      <td>6.585198</td>\n",
       "      <td>1.085120</td>\n",
       "      <td>1.658985</td>\n",
       "      <td>4.245257</td>\n",
       "      <td>1.313158</td>\n",
       "      <td>0.930872</td>\n",
       "      <td>2.093615</td>\n",
       "      <td>1.817750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9993</th>\n",
       "      <td>ENSG00000156510</td>\n",
       "      <td>HKDC1</td>\n",
       "      <td>12.752686</td>\n",
       "      <td>18.532900</td>\n",
       "      <td>4.204533</td>\n",
       "      <td>20.432585</td>\n",
       "      <td>19.322412</td>\n",
       "      <td>31.742125</td>\n",
       "      <td>15.641048</td>\n",
       "      <td>11.103851</td>\n",
       "      <td>...</td>\n",
       "      <td>32.782406</td>\n",
       "      <td>23.694282</td>\n",
       "      <td>11.978535</td>\n",
       "      <td>27.744009</td>\n",
       "      <td>6.391835</td>\n",
       "      <td>41.823995</td>\n",
       "      <td>16.527893</td>\n",
       "      <td>14.898626</td>\n",
       "      <td>18.774438</td>\n",
       "      <td>12.815931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11031</th>\n",
       "      <td>ENSG00000163520</td>\n",
       "      <td>FBLN2</td>\n",
       "      <td>17.969730</td>\n",
       "      <td>5.075141</td>\n",
       "      <td>7.463640</td>\n",
       "      <td>11.790653</td>\n",
       "      <td>2.266629</td>\n",
       "      <td>3.228069</td>\n",
       "      <td>8.960443</td>\n",
       "      <td>1.309861</td>\n",
       "      <td>...</td>\n",
       "      <td>5.110313</td>\n",
       "      <td>35.619385</td>\n",
       "      <td>0.766212</td>\n",
       "      <td>0.368642</td>\n",
       "      <td>25.445648</td>\n",
       "      <td>7.315101</td>\n",
       "      <td>6.328366</td>\n",
       "      <td>3.427936</td>\n",
       "      <td>5.668154</td>\n",
       "      <td>4.435103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12664</th>\n",
       "      <td>ENSG00000169764</td>\n",
       "      <td>UGP2</td>\n",
       "      <td>15.509306</td>\n",
       "      <td>13.800729</td>\n",
       "      <td>16.107688</td>\n",
       "      <td>11.142595</td>\n",
       "      <td>13.780305</td>\n",
       "      <td>6.701474</td>\n",
       "      <td>20.853058</td>\n",
       "      <td>18.342005</td>\n",
       "      <td>...</td>\n",
       "      <td>18.352809</td>\n",
       "      <td>12.638699</td>\n",
       "      <td>49.306538</td>\n",
       "      <td>18.514396</td>\n",
       "      <td>9.022466</td>\n",
       "      <td>11.538105</td>\n",
       "      <td>14.251787</td>\n",
       "      <td>17.314789</td>\n",
       "      <td>11.909645</td>\n",
       "      <td>13.931512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17247</th>\n",
       "      <td>ENSG00000196924</td>\n",
       "      <td>FLNA</td>\n",
       "      <td>98.506992</td>\n",
       "      <td>153.793369</td>\n",
       "      <td>48.447111</td>\n",
       "      <td>150.463581</td>\n",
       "      <td>75.349062</td>\n",
       "      <td>34.402131</td>\n",
       "      <td>64.230376</td>\n",
       "      <td>16.958706</td>\n",
       "      <td>...</td>\n",
       "      <td>60.825941</td>\n",
       "      <td>281.994106</td>\n",
       "      <td>8.634771</td>\n",
       "      <td>32.347940</td>\n",
       "      <td>231.079507</td>\n",
       "      <td>105.535774</td>\n",
       "      <td>60.471262</td>\n",
       "      <td>33.460458</td>\n",
       "      <td>89.713431</td>\n",
       "      <td>32.277014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13 rows × 420 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene_ensembl_id Gene_symbol  TCGA-AA-3867  TCGA-CA-6719  TCGA-NH-A50V  \\\n",
       "456    ENSG00000026508        CD44     49.437336     26.181544     43.585652   \n",
       "1287   ENSG00000075142         SRI     44.841315     38.171138     24.658549   \n",
       "1667   ENSG00000085733        CTTN     43.419941     31.008285     49.971597   \n",
       "2940   ENSG00000104635    SLC39A14     22.918339     16.831717     12.825033   \n",
       "4452   ENSG00000115590       IL1R2      1.729770      1.097672      0.570912   \n",
       "6093   ENSG00000129009        ISLR     69.811483     42.516880     51.739143   \n",
       "6396   ENSG00000131089     ARHGEF9      2.116260      2.724855      2.982264   \n",
       "6750   ENSG00000133392       MYH11     24.526326     60.857761      5.664525   \n",
       "8483   ENSG00000143882    ATP6V1C2      2.374979      1.542474      1.674405   \n",
       "9993   ENSG00000156510       HKDC1     12.752686     18.532900      4.204533   \n",
       "11031  ENSG00000163520       FBLN2     17.969730      5.075141      7.463640   \n",
       "12664  ENSG00000169764        UGP2     15.509306     13.800729     16.107688   \n",
       "17247  ENSG00000196924        FLNA     98.506992    153.793369     48.447111   \n",
       "\n",
       "       TCGA-AA-A01C  TCGA-AA-A02F  TCGA-AA-A03F  TCGA-AZ-4615  TCGA-AA-3854  \\\n",
       "456       25.303669     57.488870     22.468272     46.933128     56.347290   \n",
       "1287      43.792638     27.200675     15.084670     17.863711    114.546805   \n",
       "1667      43.526294     23.395513     70.448546     30.153891     44.858901   \n",
       "2940      20.000358     20.654395     12.901203     59.061575     23.314894   \n",
       "4452       0.479973      0.822586      2.869665      4.744338     31.282369   \n",
       "6093      58.313480     10.410286      8.693111     27.509687      5.299220   \n",
       "6396       2.773046      2.261290      2.707335      2.139916      2.228208   \n",
       "6750      15.126343      6.824040      3.769247      2.303767      1.568848   \n",
       "8483       4.851438      2.605318      0.851872      0.524931      0.643164   \n",
       "9993      20.432585     19.322412     31.742125     15.641048     11.103851   \n",
       "11031     11.790653      2.266629      3.228069      8.960443      1.309861   \n",
       "12664     11.142595     13.780305      6.701474     20.853058     18.342005   \n",
       "17247    150.463581     75.349062     34.402131     64.230376     16.958706   \n",
       "\n",
       "       ...  TCGA-A6-6138  TCGA-A6-2685  TCGA-QG-A5YX  TCGA-AY-A54L  \\\n",
       "456    ...     34.361535     32.096380     29.925135     32.409770   \n",
       "1287   ...     37.183063     47.035315     48.634665     60.581531   \n",
       "1667   ...     29.078803     38.652049     43.775762     32.808479   \n",
       "2940   ...     48.667234     22.855138     34.125630     22.872968   \n",
       "4452   ...      2.068797      2.393464     14.959713      0.994161   \n",
       "6093   ...     26.867733     71.743560      7.693835      1.601303   \n",
       "6396   ...      2.665840      3.260938      3.532286      3.843074   \n",
       "6750   ...     12.369897    138.054493      1.350113      0.849897   \n",
       "8483   ...      1.414512      5.900615      6.585198      1.085120   \n",
       "9993   ...     32.782406     23.694282     11.978535     27.744009   \n",
       "11031  ...      5.110313     35.619385      0.766212      0.368642   \n",
       "12664  ...     18.352809     12.638699     49.306538     18.514396   \n",
       "17247  ...     60.825941    281.994106      8.634771     32.347940   \n",
       "\n",
       "       TCGA-A6-2671  TCGA-NH-A8F8  TCGA-CM-5863  TCGA-A6-3809.2  TCGA-D5-6932  \\\n",
       "456       31.520747     50.971634     42.929245       59.984435     50.069357   \n",
       "1287      28.211444     23.956176     56.181913       25.066261     22.739080   \n",
       "1667      32.837844     42.640666     32.527568       39.002346     47.686444   \n",
       "2940       7.496100     21.732310     32.023106       25.836310     18.235208   \n",
       "4452       1.415372      0.253236      8.802575        2.729988      1.729465   \n",
       "6093      49.215278     37.554881     21.229201        2.308740     30.980162   \n",
       "6396       2.326598      2.873641      2.192778        1.684769      2.963232   \n",
       "6750      25.909564     52.773071     19.038967        0.888296     18.184051   \n",
       "8483       1.658985      4.245257      1.313158        0.930872      2.093615   \n",
       "9993       6.391835     41.823995     16.527893       14.898626     18.774438   \n",
       "11031     25.445648      7.315101      6.328366        3.427936      5.668154   \n",
       "12664      9.022466     11.538105     14.251787       17.314789     11.909645   \n",
       "17247    231.079507    105.535774     60.471262       33.460458     89.713431   \n",
       "\n",
       "       TCGA-AA-3675  \n",
       "456       59.563262  \n",
       "1287      36.772185  \n",
       "1667      21.455800  \n",
       "2940      27.093669  \n",
       "4452       1.220621  \n",
       "6093      16.907314  \n",
       "6396       2.286335  \n",
       "6750       1.436682  \n",
       "8483       1.817750  \n",
       "9993      12.815931  \n",
       "11031      4.435103  \n",
       "12664     13.931512  \n",
       "17247     32.277014  \n",
       "\n",
       "[13 rows x 420 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FPKM_tumour_iso"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Perform filter on normal FPKM \n",
    "\n",
    "FPKM_normal_iso = FPKM_normal.loc[FPKM_normal['Gene_symbol'].isin(isoform_genes_FPKM)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "#No filter for these isoform genes in the MAF, Hugo_symbol col in the tumour and normal DFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#First Check if any present in isoform_switches not found in MAF\n",
    "#Could be due to differences in gene label => add manualy to MAF_iso if so\n",
    "\n",
    "set(isoform_switches['Gene_symbol']) - set(MAF_VEP_protein_tumour['Hugo_Symbol']) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>src_vcf_id</th>\n",
       "      <th>tumor_bam_uuid</th>\n",
       "      <th>normal_bam_uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>GDC_FILTER</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>MC3_Overlap</th>\n",
       "      <th>GDC_Validation_Status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Hugo_Symbol, Entrez_Gene_Id, Center, NCBI_Build, Chromosome, Start_Position, End_Position, Strand, Variant_Classification, Variant_Type, Reference_Allele, Tumor_Seq_Allele1, Tumor_Seq_Allele2, dbSNP_RS, dbSNP_Val_Status, Tumor_Sample_Barcode, Matched_Norm_Sample_Barcode, Match_Norm_Seq_Allele1, Match_Norm_Seq_Allele2, Tumor_Validation_Allele1, Tumor_Validation_Allele2, Match_Norm_Validation_Allele1, Match_Norm_Validation_Allele2, Verification_Status, Validation_Status, Mutation_Status, Sequencing_Phase, Sequence_Source, Validation_Method, Score, BAM_File, Sequencer, Tumor_Sample_UUID, Matched_Norm_Sample_UUID, HGVSc, HGVSp, HGVSp_Short, Transcript_ID, Exon_Number, t_depth, t_ref_count, t_alt_count, n_depth, n_ref_count, n_alt_count, all_effects, Allele, Gene, Feature, Feature_type, One_Consequence, Consequence, cDNA_position, CDS_position, Protein_position, Amino_acids, Codons, Existing_variation, ALLELE_NUM, DISTANCE, TRANSCRIPT_STRAND, SYMBOL, SYMBOL_SOURCE, HGNC_ID, BIOTYPE, CANONICAL, CCDS, ENSP, SWISSPROT, TREMBL, UNIPARC, RefSeq, SIFT, PolyPhen, EXON, INTRON, DOMAINS, GMAF, AFR_MAF, AMR_MAF, ASN_MAF, EAS_MAF, EUR_MAF, SAS_MAF, AA_MAF, EA_MAF, CLIN_SIG, SOMATIC, PUBMED, MOTIF_NAME, MOTIF_POS, HIGH_INF_POS, MOTIF_SCORE_CHANGE, IMPACT, PICK, VARIANT_CLASS, TSL, HGVS_OFFSET, PHENO, MINIMISED, ...]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 120 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for prescence of alternative symbols ..\n",
    "\n",
    "MAF_VEP_protein_tumour[MAF_VEP_protein_tumour['Hugo_Symbol']=='']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create list of iso_form genes to filter for, including any identified above with alternative names\n",
    "\n",
    "isoform_genes_MAF = list(isoform_switches['Gene_symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MAF tumour filter for isoform switch genes\n",
    "\n",
    "MAF_tumour_iso = MAF_VEP_protein_tumour.loc[MAF_VEP_protein_tumour['Hugo_Symbol'].isin(isoform_genes_MAF)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MAF normal filter for isoform switch genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "MAF_normal_iso = MAF_VEP_protein_normal.loc[MAF_VEP_protein_normal['Hugo_Symbol'].isin(isoform_genes_MAF)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FLNA        36\n",
       "MYH11       27\n",
       "HKDC1       17\n",
       "FBLN2       13\n",
       "UGP2         9\n",
       "IL1R2        7\n",
       "ARHGEF9      7\n",
       "ISLR         7\n",
       "ATP6V1C2     4\n",
       "SLC39A14     4\n",
       "CTTN         2\n",
       "SRI          2\n",
       "CD44         2\n",
       "Name: Hugo_Symbol, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MAF_tumour_iso['Hugo_Symbol'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FLNA        3\n",
       "MYH11       2\n",
       "IL1R2       1\n",
       "FBLN2       1\n",
       "SLC39A14    1\n",
       "HKDC1       1\n",
       "Name: Hugo_Symbol, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MAF_normal_iso['Hugo_Symbol'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Hugo_Symbol  Transcript_ID  \n",
       "ABLIM3       ENST00000309868     4\n",
       "ADAMTS12     ENST00000504830     8\n",
       "ATXN3        ENST00000532032     2\n",
       "BMP1         ENST00000306385     3\n",
       "CASC4        ENST00000299957     4\n",
       "CCND3        ENST00000372991     1\n",
       "CEP104       ENST00000378230     3\n",
       "DST          ENST00000312431    32\n",
       "DUT          ENST00000331200     1\n",
       "EHBP1        ENST00000263991     8\n",
       "FBLN2        ENST00000295760     2\n",
       "GSPT1        ENST00000420576     5\n",
       "INHBA        ENST00000242208     4\n",
       "INSR         ENST00000302850    10\n",
       "ISLR         ENST00000249842     4\n",
       "LRRFIP2      ENST00000336686     2\n",
       "LSP1         ENST00000311604     3\n",
       "MCF2L        ENST00000375608     7\n",
       "MPRIP        ENST00000341712     3\n",
       "MYL6         ENST00000548293     2\n",
       "PCGF5        ENST00000336126     2\n",
       "PPP2R1B      ENST00000311129     1\n",
       "PRMT2        ENST00000355680     2\n",
       "REPS1        ENST00000450536     3\n",
       "SERPINA1     ENST00000355814     6\n",
       "SLK          ENST00000369755     5\n",
       "SNRNP70      ENST00000598441     2\n",
       "TANK         ENST00000259075     2\n",
       "TMEM126B     ENST00000358867     1\n",
       "TMEM201      ENST00000340381     1\n",
       "ZNF207       ENST00000321233     3\n",
       "ZNF436       ENST00000314011     2\n",
       "ZRANB2       ENST00000370920     2\n",
       "Name: Transcript_ID, dtype: int64"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#For each gene there is only one associated transcript_ID \n",
    "#=> cant filter for Transcript id suplied with isoform switches\n",
    "\n",
    "MAF_tumour_iso.groupby('Hugo_Symbol')['Transcript_ID'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
